.gitlab/build_tioga.yml

```diff
@@ -38,8 +38,20 @@ tioga-rocmcc_6_2_1_hip-src:
   variables:
     COMPILER: "rocmcc@6.2.1_hip"
     HOST_CONFIG: "tioga-toss_4_x86_64_ib_cray-${COMPILER}.cmake"
+    NUM_NODES: "1"
   extends: .src_build_on_tioga
 
+tioga-rocmcc_6_3_1_hip-gpu-aware-src:
+  variables:
+    COMPILER: "rocmcc@6.3.1_hip"
+    HOST_CONFIG: "tioga-toss_4_x86_64_ib_cray-${COMPILER}.cmake"
+    ALLOC_TIME: "60"
+    NUM_NODES: "2"
+  extends: .src_build_on_tioga
+  before_script:
+    - module load cmake/3.21.1
+    - export MPICH_GPU_SUPPORT_ENABLED=1
```

Comment:
This is the CI job I added to replicate the failure. Notably this does require more than one node allocation to reproduce the error.

My concern is there are only 4 nodes in the special partition CI queue: https://lc.llnl.gov/confluence/spaces/GITLAB/pages/785285769/Special+partitions+for+CI+batch+jobs, so adding this job will make an entire CI run take up most of the queue.
I am leaning towards removing the additional CI job from this PR for that reason.

Reply:
We may be able to get LC to bump the number of CI nodes.